memstore/table_shard.go

```diff
@@ -71,19 +71,27 @@ func NewTableShard(schema *TableSchema, metaStore metastore.MetaStore,
 	return tableShard
 }
 
-// SetDiskPurgeEnabled set whether disk purge is enabled
-func (shard *TableShard) SetDiskPurgeEnabled(enabled bool) {
-	var disabled uint32 = 0
-	if !enabled {
-		disabled = 1
-	}
-	atomic.StoreUint32(&shard.diskPurgeDisabled, disabled)
+// PinForPeerDataTransfer pin disk file from being purged for peer data transfer
+// User should unpin after data transfer is done
+func (shard *TableShard) PinForPeerDataTransfer() {
+	pins := atomic.AddInt32(&shard.pinnedDiskFileUsers, 1)
+	utils.GetReporter(shard.Schema.Schema.Name, shard.ShardID).GetGauge(utils.PinnedDiskFileUsers).
+		Update(float64(pins))
+}
+
+// DoneWithPeerDataTransfer unpin disk file after data transfer is done
+func (shard *TableShard) DoneWithPeerDataTransfer() {
+	pins := atomic.AddInt32(&shard.pinnedDiskFileUsers, -1)
+	utils.GetReporter(shard.Schema.Schema.Name, shard.ShardID).GetGauge(utils.PinnedDiskFileUsers).
+		Update(float64(pins))
 }
 
 // IsDiskPurgeEnabled check whether disk purge is enabled
 func (shard *TableShard) IsDiskPurgeEnabled() bool {
```

Comment:
how will this function be used? The check may be incorrect if it is enabled/disabled after this function call

Reply:
This function will be used before archiving disk purge step